### ISSUE TITLE
suggestion: use args for ur_description

### ISSUE THREAD

**Issue by ThomasTimm** [ros-industrial/universal_robot#234]
Right now we have two robot.urdf.xacro files for each robot; one with limited joint movements and another without.

As Xacro has support for arg (just like launch files), I think it would make more sense to just have one file, and then pass the limited arg from the launch file to the xacro file.

The same could be applied for the prefix tag, instead of ppl having to make a copy of the robot.urdf.xacro file if they are using two arm and thus needs to specify a prefix.

**Comment by ThomasTimm** [comment]
The part concerning the prefix will only be possible once we move to jade, as proper conditional evaluation is needed.

**Comment by gavanderhoorn** [comment]
Yeah. And that is exactly why the current implementation is as it is right now ..

**Comment by ThomasTimm** [comment]
The issue is with handling empty strings; for some reason xacro is really bad at handling that. The 'limited' parameter will always be either true or false, so that can still be changed now.

Also, I guess ppl would make a new robot.urdf.xacro file anyway if they were using two arms - otherwise the two arms would be located on top of each other. So I guess the latter part about making the prefix tag an arguement is irrelevant.

**Comment by gavanderhoorn** [comment]
Related: #144.

Yaml with substitution args or even the new [yaml support in xacro](http://wiki.ros.org/xacro#YAML_support) could be approaches to consider here.

Configuring the xacro macro from a yaml file completely would have my preference.

**Comment by dhled** [comment]
Didn't finish it yet. Will try to push it this week.

**Comment by dhled** [comment]
Proposition of solution: https://github.com/ros-industrial/universal_robot/pull/371
